master/pkg/tasks/task_trial.go

```diff
@@ -13,20 +13,21 @@ import (
 	"github.com/determined-ai/determined/master/pkg/model"
 	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
 	"github.com/determined-ai/determined/master/pkg/ssh"
-	"github.com/determined-ai/determined/master/pkg/workload"
 )
 
 // TrialSpec is a description of a task for running a trial container.
 type TrialSpec struct {
 	Base TaskSpec
 
-	ExperimentConfig    expconf.ExperimentConfig
-	ModelDefinition     archive.Archive
-	HParams             map[string]interface{}
-	TrialSeed           uint32
-	LatestCheckpoint    *model.Checkpoint
-	InitialWorkload     workload.Workload
-	WorkloadManagerType model.WorkloadManagerType
+	ExperimentID     int
+	TrialID          int
+	TrialRunID       int
+	ExperimentConfig expconf.ExperimentConfig
+	ModelDefinition  archive.Archive
+	HParams          map[string]interface{}
+	TrialSeed        uint32
+	LatestCheckpoint *model.Checkpoint
+	LatestBatch      int
```

Comment:
it is part of the checkpoint, there is some logic right above this to set it if it was our trials checkpoint and ignore if it was a warm start above. i imagine when this information in the checkpoint becomes metadata instead of required we'll refactor this to be a bit nicer.

Reply:
We shouldn't override this and ignore the metadata `latest_batch` at all. I don't think we should ever try to do this. If it's a warm start, we should just have a separate field called `warm_start` to tell harness to ignore the `latest_batch` saved in the training loop checkpoint. The point is that the checkpoint should contain everything that starting a training loop needs.

Reply:
that's fine with me, but they're exactly the same information encoded in a different way, i dont see it as a huge deal. i suppose it's better to give the user a checkpoint and tell them they're warm starting rather than having the master dictate a batch index to the harness. @rb-determined-ai thoughts (i can just update go/python if you're game)?

Reply:
As I said [here](https://github.com/determined-ai/determined/pull/2771#discussion_r678679230):

> 

> While I agree with you in general, it leads to an obnoxious initialization dependency that I didn't want to have to solve yet.
>
> I would like us to have the ability to set and fetch generic metadata on checkpoints. When we have that, we would probably want to tag training checkpoints with total_batches, and we would get it from there. At that time we would remove DET_LATEST_BATCH from the environment.
>
> I don't think fixing this before then buys us very much at all.

I'm happy to create a ticket for removing the concept of latest batch before we land generic API, if @shiyuann feels that is sufficient.  In the mean time, it seems like a pretty harmless internal detail.

Reply:
Sounds good! Let's create a ticket for it. I think it's important because if we were to support restore from the middle of the training loop in local training mode. It will be useful to make the checkpoint self-contained.